# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/xul/nsXULPopupManager.h
# Commit: 2424d5c30267
# Full Hash: 2424d5c30267f67ab2b28119d75d89f0441b30e0
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2024-01-23 05:36:48
# Regressor Bug: 1833814
# File Overlap Count: 1
# Description:
#   Bug 1833814 - change nsIWidget's cursor logic, r=emilio,win-reviewers,mhowell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195549
# ==============================================================================

diff -r 94f46aad00df -r 2424d5c30267 layout/xul/nsXULPopupManager.h
--- a/layout/xul/nsXULPopupManager.h	Mon Jan 22 20:10:54 2024 +0000
+++ b/layout/xul/nsXULPopupManager.h	Mon Jan 22 20:42:50 2024 +0000
@@ -23,6 +23,7 @@
 #include "nsIObserver.h"
 #include "nsThreadUtils.h"
 #include "mozilla/Attributes.h"
+#include "mozilla/FunctionRef.h"
 #include "mozilla/widget/InitData.h"
 #include "mozilla/widget/NativeMenu.h"
 
@@ -551,6 +552,13 @@
   void UpdatePopupPositions(nsRefreshDriver* aRefreshDriver);
 
   /**
+   * Get the first nsMenuChainItem that is matched by the matching callback
+   * function provided.
+   */
+  nsMenuChainItem* FirstMatchingPopup(
+      mozilla::FunctionRef<bool(nsMenuChainItem*)> aMatcher) const;
+
+  /**
    * Enable or disable anchor following on the popup if needed.
    */
   void UpdateFollowAnchor(nsMenuPopupFrame* aPopup);
@@ -733,6 +741,9 @@
     return GetRollupItem(RollupKind::Menu);
   }
 
+  // Add the chain item to the chain and update mPopups to point to it.
+  void AddMenuChainItem(mozilla::UniquePtr<nsMenuChainItem>);
+
   // Removes the chain item from the chain and deletes it.
   void RemoveMenuChainItem(nsMenuChainItem*);
 